### PR TITLE
Fix/approval deposit

### PR DESCRIPTION
**Pull request by valentinnavalos**
Remaining tasks:
- Wait for the tx confimation.
- Make the tx handler for use at deposit & publish modals.

### CODE REVIEW

File: packages/web-app/src/containers/transactionModals/DepositModal.tsx [hunk 1]
```diff
@@ -81,28 +91,34 @@ const DepositModal: React.FC = () => {
       const amount = Number(input.amount) * Math.pow(10, tokenInfo.decimals);
       const allowance = await tokenAllowance(input.tokenAddress);
       if (allowance < amount) {
-        setDepositProcessState(TransactionState.APPROVING);
-        approve(input.tokenAddress, amount); // in promise
-        setDepositProcessState(TransactionState.LOADING);
+        try {
+          await approve(input.tokenAddress, amount);
+          setDepositProcessState(TransactionState.APPROVING);
```
Comment (ialberquilla) [inline packages/web-app/src/containers/transactionModals/DepositModal.tsx]: I think this should go before approved, and then once approved move to deposit right?

**Comment by ialberquilla** [comment]
Is this finished?  @valentinnavalos

**Comment by valentinnavalos** [comment]
> Is this finished? @valentinnavalos

Not yet.

**Comment by valentinnavalos** [comment]
@ialberquilla i just added the `waitForTx` fn, but i got "execution reverted: ERC20: transfer amount exceeds balance" error. I have enough balance, idk.

**Comment by ialberquilla** [comment]
> @ialberquilla i just added the `waitForTx` fn, but i got "execution reverted: ERC20: transfer amount exceeds balance" error. I have enough balance, idk.

ok, will take a look to that

**Review by ialberquilla — APPROVED** [review]
It is working perfectly for me 🚀! we can merge. Maybe you don't have funds in this specific USDC account

**Pull request merged**